src/test/xaddress_tests.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2021 Logos Foundation
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <addresses/xaddress.h>
+#include <random.h>
+#include <test/util/setup_common.h>
+#include <util/strencodings.h>
+#include <util/vector.h>
+
+#include <algorithm>
+#include <boost/test/unit_test.hpp>
+#include <functional>
+#include <random>
+#include <string>
+
+/** All alphanumeric characters except for "0", "I", "O", and "l" */
+const std::string pszBase58 =
+    "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+
+BOOST_FIXTURE_TEST_SUITE(xaddress_tests, BasicTestingSetup)
+
+BOOST_AUTO_TEST_CASE(raw_encode_decode_succeeds) {
+    // First create an instance of an engine.
+    std::random_device rnd_device;
+    // Specify the engine and distribution.
+    std::mt19937 mersenne_engine{rnd_device()}; // Generates random integers
```

Comment:
We can just use the `MMIXLinearCongruentialGenerator` in `<lgc.h>`; tests don't have to be cryptographically secure

Reply:
Sure

Reply:
Actually, that doesn't generate uniform data, right? This isn't trying to span some range, it's just a random data payload. What's wrong with this? This already isn't cryptographically secure anyways.